### PR TITLE
feat(helm): Introduce priorityClassName system-node-critical

### PR DESCRIPTION
**Pull request by ffuerste**
# What does this PR do?

Introducing in the helm charts of all NRI plugins an optional boolean field for setting `priorityClassName: system-node-critical`.

As suggest [here](https://github.com/containers/nri-plugins/issues/194#issuecomment-1833377246) the `priorityClassName` is introduced as an optional helm flag analog to `patchRuntimeConfig`. However, in this way the feature can only be enabled with no option to define a custom `priorityClassName`. 

## Related issues

Resolves containers/nri-plugins#194

### CODE REVIEW

**Review by marquiz — COMMENTED** [review]
Thanks for the patch @ffuerste. This totally makes sense.

I'm just wondering should be the desired default be `true`, for at least some of the plugins 🧐 If a resource policy plugin (like balloons or topology-aware) is deployed I think it is kinda critical and should not be evicted (thus default to `true`). WDYT? @klihub @fmuyassarov @askervin @kad

**Review by askervin — APPROVED** [review]
I think this is a very nice improvement.

I'm fine both merging this PR as it is, or with some/all existing plugins set as `system-node-critical` by default.

Maybe I'd even prefer taking this PR is as it is, as it only enables configuring the priorityClassName but does not change the default behavior. And then have a separate commit that changes the default behavior for chosen plugins.

**Comment by fmuyassarov** [comment]
> Thanks for the patch @ffuerste. This totally makes sense.
> 
> I'm just wondering should be the desired default be `true`, for at least some of the plugins 🧐 If a resource policy plugin (like balloons or topology-aware) is deployed I think it is kinda critical and should not be evicted (thus default to `true`). WDYT? @klihub @fmuyassarov @askervin @kad

I don't have a strong opinion about that to be frank. I'm fine with the current state as well.

**Comment by kad** [comment]
I think it makes sense.

**Review by klihub — APPROVED** [review]
LGTM.

**Comment by klihub** [comment]
> Thanks for the patch @ffuerste. This totally makes sense.
> 
> I'm just wondering should be the desired default be `true`, for at least some of the plugins 🧐 If a resource policy plugin (like balloons or topology-aware) is deployed I think it is kinda critical and should not be evicted (thus default to `true`). WDYT? @klihub @fmuyassarov @askervin @kad

I'm fine with the default either way for the current plugins.

File: deployment/helm/balloons/values.yaml [hunk 1]
```diff
@@ -42,6 +42,10 @@ resources:
 
 nri:
   patchRuntimeConfig: false
+  # NRI plugins should be considered as part of the container runtime.
+  # Therefore, adding the system-node-critical priority class to the DaemonSet,
+  # could mitigate potential risk in a running system under load.
+  setPodPriorityClassNodeCritical: false
```
Comment (marquiz) [inline deployment/helm/balloons/values.yaml:48]: Two suggestions:

1. Don't put the new config field under `nri:` but on the top level. I think this is better in line with the other pod level settings like `toleration` and `nodeSelector`
2. Streamline the field name a bit, drop the "set" from the beginning

```suggestion
podPriorityClassNodeCritical: false
```

File: deployment/helm/balloons/README.md [hunk 2]
```diff
@@ -98,6 +98,7 @@ customize with their own values, along with the default values.
 | `hostPort`               | 8891                                                                                                                          | metrics port to expose on the host                   |
 | `config`                 | see [helm chart values](tree:/deployment/helm/balloons/values.yaml) for the default configuration                       | plugin configuration data                            |
 | `nri.patchRuntimeConfig` | false                                                                                                                         | enable NRI in containerd or CRI-O                    |
+| `nri.setPodPriorityClassNodeCritical` | false                                                                                                                         | enable [marking Pod as node critical](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical)                       |
```
Comment (marquiz) [inline deployment/helm/balloons/README.md:101]: Accompaniying my other comment:

```suggestion
| `podPriorityClassNodeCritical` | false                                                                                                                         | enable [marking Pod as node critical](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/#marking-pod-as-critical)                       |
```

...and ditto all charts

**Review by marquiz — CHANGES_REQUESTED** [review]
Hold until comments are resolved

File: deployment/helm/balloons/values.yaml (hunk 1 above)
Comment (ffuerste) [reply deployment/helm/balloons/values.yaml]: @marquiz I have updated the helm charts as suggested. I have added the "new" value `podPriorityClassNodeCritical` at the end of the `values.yaml` files to indicate that it is similar to the `toleration` and `nodeSelector` values. Please check again. Thank you very much!

File: deployment/helm/balloons/README.md (hunk 2 above)
Comment (ffuerste) [reply deployment/helm/balloons/README.md]: @marquiz I have updated the helm charts as suggested. I've added the description of the "new" value `podPriorityClassNodeCritical` at the end of the helm value table, reflecting its position in the `values.yaml` files. I am resolving this thread in favour of the first thread.

**Review by marquiz — APPROVED** [review]
Thanks @ffuerste. LGTM on my part

**Review by fmuyassarov — APPROVED** [review]

**Pull request merged**